src/main/java/com/zendesk/maxwell/schema/SchemaStoreSchema.java

```diff
@@ -119,21 +122,43 @@ public static void upgradeSchemaStoreSchema(Connection c, String schemaDatabaseN
 			performAlter(c, "alter table `bootstrap` modify column inserted_rows bigint unsigned not null default 0");
 		}
 
-		if ( !getTableColumns("schemas", c).containsKey("charset")) {
+
+		HashMap<String, String> schemaColumns = getTableColumns("schemas", c);
+		if ( !schemaColumns.containsKey("charset")) {
 			String[] charsetTables = { "schemas", "databases", "tables", "columns" };
 			for ( String table : charsetTables ) {
 				performAlter(c, "alter table `" + table + "` change `encoding` `charset` varchar(255)");
 			}
 		}
 
-		if ( !getTableColumns("schemas", c).containsKey("base_schema_id"))
+		if ( !schemaColumns.containsKey("base_schema_id"))
 			performAlter(c, "alter table `schemas` add column base_schema_id int unsigned NULL default NULL after binlog_position");
 
-		if ( !getTableColumns("schemas", c).containsKey("deltas"))
+		if ( !schemaColumns.containsKey("deltas"))
 			performAlter(c, "alter table `schemas` add column deltas mediumtext charset 'utf8' NULL default NULL after base_schema_id");
 
-		if ( !getTableColumns("schemas", c).containsKey("version")) {
+		if ( !schemaColumns.containsKey("version")) {
 			performAlter(c, "alter table `schemas` add column `version` smallint unsigned not null default 0 after `charset`");
 		}
+
+		if ( !getTableColumns("positions", c).containsKey("client_id") ) {
+			performAlter(c, "alter table `positions` add column `client_id` varchar(255) charset 'latin1' not null default 'maxwell'");
+			performAlter(c, "alter table `positions` add column drop primary key, add primary key(`server_id`, `client_id`)");
+		}
+
+		if ( !schemaColumns.containsKey("position_sha") ) {
+			performAlter(c, "alter table `schemas` add column `position_sha` char(40) charset 'latin1' null default null, add unique index(`position_sha`)");
```

Comment:
it's a hex-digest of a SHA.  doing it as `latin` makes the size of the unique index managable, and there's no chance of high-bit utf8 here.